libsolidity/ast/Types.cpp

```diff
@@ -561,6 +812,75 @@ shared_ptr<IntegerType const> IntegerConstantType::integerType() const
 		);
 }
 
+shared_ptr<FixedPointType const> RationalNumberType::fixedPointType() const
+{
+	//do calculations up here
```

Comment:
here's my patched up version. For some reason it is effecting my unary operation test with the negative...

```
shared_ptr<FixedPointType const> RationalNumberType::fixedPointType() const
{
	bool negative = (m_value < 0);
	unsigned fractionalBits = 0;
	unsigned integerBits = bytesRequired(wholeNumbers()) * 8;
	rational value = m_value;
	bigint l = bigint(1) << 256;
	rational maxValue = rational(l);
	if (!negative)
		maxValue -= 1;
	else
		value = -value;
	cout << "Max value: " << maxValue << endl;
	while (value * 0x100 <= maxValue && value.denominator() != 1 && fractionalBits < 256 - integerBits)
	{
		value *= 0x100;
		fractionalBits += 8;
	}

	if (value > maxValue)
	{
		cout << "value > max value " << endl;
		return shared_ptr<FixedPointType const>();
	}
	bigint v = value.denominator() / value.numerator();
	if (negative)
		v = -v;
	// u256(v) is the actual value that will be put on the stack
	// From here on, very similar to integerType()
	//if (negative) // convert to positive number of same bit requirements
	//	value = ((0 - value) - 1) << 1;
	if (value > u256(-1))
	{
		cout << "Too large of a number" << endl;
		return shared_ptr<FixedPointType const>();
	}
	
	//solAssert(integerBits >= fractionalBits, "Invalid bit requirement calculation.");
	//@todo special handling for integerBits == 0 && fractionalBits == 0?
	return make_shared<FixedPointType>(
		integerBits, fractionalBits,
		negative ? FixedPointType::Modifier::Signed : FixedPointType::Modifier::Unsigned
	);
}
```